template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/efm/EfmRoleConfigProviderTest.java

```diff
@@ -0,0 +1,71 @@
+package com.sequenceiq.cloudbreak.cmtemplate.configproviders.efm;
+
+import static com.sequenceiq.cloudbreak.cmtemplate.configproviders.ConfigUtils.config;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import com.cloudera.api.swagger.model.ApiClusterTemplateConfig;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.common.DatabaseVendor;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.database.base.DatabaseType;
+import com.sequenceiq.cloudbreak.cmtemplate.CmTemplateProcessor;
+import com.sequenceiq.cloudbreak.domain.view.RdsConfigWithoutCluster;
+import com.sequenceiq.cloudbreak.template.TemplatePreparationObject;
+import com.sequenceiq.cloudbreak.template.filesystem.TemplateCoreTestUtil;
+import com.sequenceiq.cloudbreak.template.views.BlueprintView;
+import com.sequenceiq.cloudbreak.util.FileReaderUtils;
+
+@RunWith(MockitoJUnitRunner.class)
+public class EfmRoleConfigProviderTest {
+
+    private static final String CONNECTION_URL = "jdbc:postgresql://testhost:5432/efm";
+
+    private static final String USER_NAME = "efm_server_user";
+
+    private static final String USER_PASSWORD = "efm_server_password";
+
+    private final EfmRoleConfigProvider underTest = new EfmRoleConfigProvider();
```

Comment:
Even if we expect the class to be stateless, it is safer to instantiate and assign `underTest` for each test case execution, i.e. using a `@BeforeEach setUp()` method (and also dropping `final` from the field).

A similar comment would also apply to `EfmRdsConfigProviderTest`.

Reply:
I made the changes you requested junit4 and junit5 is used in the two classes respectively so BeforeEach and Before was utilised.

Reply:
JUnit 4 should be avoided; please see my below comment as well. :smile: